### PR TITLE
Small tweaks to not allocate objects for lambda closures

### PR DESCRIPTION
**Pull request by bbakerman**
more bottom of the barrel improvements

Capto_Capture 2024-03-08_10-00-53_am

Capto_Capture 2024-03-08_10-01-08_am

These ones are easy ish.

other closures are not so easy

### CODE REVIEW

File: src/main/java/graphql/execution/ExecutionStepInfo.java [hunk 1]
```diff
@@ -299,11 +299,8 @@ public Builder path(ResultPath resultPath) {
             return this;
         }
 
-        public Builder arguments(Supplier<Map<String, Object>> arguments) {
-            this.arguments = () -> {
-                Map<String, Object> map = arguments.get();
-                return map == null ? ImmutableMapWithNullValues.emptyMap() : ImmutableMapWithNullValues.copyOf(map);
-            };
+        public Builder arguments(Supplier<ImmutableMapWithNullValues<String, Object>> arguments) {
```
Comment (bbakerman) [inline src/main/java/graphql/execution/ExecutionStepInfo.java:302]: Technically this is a breaking change

I was going to leave the old and deprecate it and make a new one - but why - we dont really expect any one to build ESI objects and hence they can fix it up

the only caller of this is one place in the ES code

If you object - I can leave the old - I actually had that and then say "nah yeah nah"

File: src/main/java/graphql/execution/ExecutionStrategy.java [hunk 2]
```diff
@@ -1024,4 +1016,25 @@ protected ExecutionStepInfo createExecutionStepInfo(ExecutionContext executionCo
...
+                    executionContext.getCoercedVariables(),
+                    executionContext.getGraphQLContext(),
+                    executionContext.getLocale());
+
+            return ImmutableMapWithNullValues.copyOf(resolvedValues);
+        };
+        argumentValues = FpKit.intraThreadMemoize(argValuesSupplier);
+        return argumentValues;
+    }
+
```
Comment (bbakerman) [inline src/main/java/graphql/execution/ExecutionStrategy.java:1039]: made a method - they are all the rage these days

File: src/main/java/graphql/schema/PropertyFetchingImpl.java [hunk 3]
```diff
@@ -66,7 +66,7 @@ private static final class CachedLambdaFunction {
         }
     }
 
-    public Object getPropertyValue(String propertyName, Object object, GraphQLType graphQLType, boolean dfeInUse, Supplier<Object> singleArgumentValue) {
+    public Object getPropertyValue(String propertyName, Object object, GraphQLType graphQLType, boolean dfeInUse, Supplier<?> singleArgumentValue) {
```
Comment (bbakerman) [inline src/main/java/graphql/schema/PropertyFetchingImpl.java:69]: This then allows the supplier to come in without an adaptor

File: src/main/java/graphql/schema/PropertyDataFetcherHelper.java [hunk 4]
```diff
@@ -13,13 +13,14 @@
 public class PropertyDataFetcherHelper {
 
     private static final PropertyFetchingImpl impl = new PropertyFetchingImpl(DataFetchingEnvironment.class);
+    private static final Supplier<Object> ALWAYS_NULL = () -> null;
 
     public static Object getPropertyValue(String propertyName, Object object, GraphQLType graphQLType) {
-        return impl.getPropertyValue(propertyName, object, graphQLType, false, () -> null);
+        return impl.getPropertyValue(propertyName, object, graphQLType, false, ALWAYS_NULL);
```
Comment (bbakerman) [inline src/main/java/graphql/schema/PropertyDataFetcherHelper.java:19]: in practice this was a singleton - now its more clear it is

File: src/main/java/graphql/execution/ExecutionStrategy.java (hunk 2 above)
Comment (dondonz) [reply src/main/java/graphql/execution/ExecutionStrategy.java:1039]: You're always up to date with the latest trends!

**Pull request merged**